codex-ui/src/vue/components/popup/Popup.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div
+    v-show="isOpen"
+    :class="$style['popup']"
+  >
+    <div
+      ref="popupEl"
+      :class="$style['popup__container']"
+    >
+      <component
+        :is="content.component"
+        v-if="content"
+        v-bind="content.props"
+      />
+    </div>
+    <Icon
+      v-if="hasCloseButton"
+      :class="$style['popup__icon']"
+      name="Cross"
+      @click="hidePopup"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted } from 'vue';
+import Icon from '../icon/Icon.vue';
+import { usePopup } from './usePopup';
+import { onClickOutside } from '@vueuse/core';
+
+/**
+ * Reference to the popup element
+ * Used to bind click-outside event
+ */
+const popupEl = ref<HTMLDivElement | null>(null);
+
+const {
+  isOpen,
+  hidePopup,
+  content,
+} = usePopup();
+
+/**
+ * Close the popup when clicking outside of popup container
+ */
+onClickOutside(popupEl, hidePopup);
+
+/**
+ * Close the popup when esc was pressed
+ *
+ * @param event - the event object representing the keyboard event
+ * @param event.key - the property of the event object that holds the value of the pressed key
+ */
+const closeOnEsc = (event: { key: string }) => {
+  if (event.key === 'Escape') {
+    hidePopup();
+  }
+};
+
+withDefaults(
+  defineProps<{
+    /**
+     * If true, a button to close the popup appears
+     */
+    hasCloseButton?: boolean;
+  }>(),
+  {
+    hasCloseButton: true,
+  }
+);
+
+onMounted(() => {
+  document.addEventListener('keydown', closeOnEsc);
+});
+
+onUnmounted(() => {
+  document.removeEventListener('keydown', closeOnEsc);
+});
+
+</script>
+
+<style module>
+
+.popup {
+  z-index: var(--z-popup);
+  inset: 0;
+  display: grid;
+  grid-template-columns: auto auto;
+  position: fixed;
+  align-content: center;
+  gap: var(--spacing-l);
+  justify-content: center;
+  align-items: start;
+  background-color: rgba(0, 0, 0, 0.49);
+  align-items: baseline;
+  color: var(--base--text);
+
+  &__container {
+    gap: var(--spacing-ml);
+    padding: var(--spacing-l);
+    border-radius: var(--radius-ml);
+    background-color: var(--base--bg-primary);
+    box-shadow: inset 0 0 0 var(--delimiter-height) var(--base--border);
+  }
+
+  &__icon {
```

Comment:
The close button is displayed lower than it should be.

<img width="494" alt="image" src="https://github.com/user-attachments/assets/ec560a85-f622-44f8-8ece-acfea6593be5">

<img width="491" alt="image" src="https://github.com/user-attachments/assets/bc9421c4-ae04-4d89-8642-49b960de91ab">

Reply:
also, please add hover to it. (Making icon and border color of `--base--text`)

Reply:
> The close button is displayed lower than it should be.
> 
> <img alt="image" width="494" src="https://private-user-images.githubusercontent.com/3684889/365144433-ec560a85-f622-44f8-8ece-acfea6593be5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NTUyNjAsIm5iZiI6MTcyNTY1NDk2MCwicGF0aCI6Ii8zNjg0ODg5LzM2NTE0NDQzMy1lYzU2MGE4NS1mNjIyLTQ0ZjgtOGVjZS1hY2ZlYTY1OTNiZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMjAzNjAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDU5MTMyNjI1YjVkMjdjYTg0MzFmZDE2NWY1MmY3MzRkOTkzZGYzMDU4MzMzNDM5ZmQ0OWY4MzI4OTBiNmIyOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PpZvhr08u-Mfi0x_L6Infkwia-LnAJOdrKbSVb3bBcg"> <img alt="image" width="491" src="https://private-user-images.githubusercontent.com/3684889/365144485-bc9421c4-ae04-4d89-8642-49b960de91ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NTUyNjAsIm5iZiI6MTcyNTY1NDk2MCwicGF0aCI6Ii8zNjg0ODg5LzM2NTE0NDQ4NS1iYzk0MjFjNC1hZTA0LTRkODktODY0Mi00OWI5NjBkZTkxYWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMjAzNjAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzM5ODkyZDY2ZDI5YWRlNThhNjFhYWE2MDM1MzY5YTE5NTAwMDY5ZmU2M2NhNTkzMzdiNDcxNjcwMDAwOGU4MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.WSCJ_OxK7LBW7rNXdTOmttuOblHaKv5fL0QykL2xZDA">

Currently it is displayed using the `align-items: baseline;` property in `popup` style
What is the best property to use instead?

Reply:
> The close button is displayed lower than it should be.
> 
> <img alt="image" width="494" src="https://private-user-images.githubusercontent.com/3684889/365144433-ec560a85-f622-44f8-8ece-acfea6593be5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NTU3ODQsIm5iZiI6MTcyNTY1NTQ4NCwicGF0aCI6Ii8zNjg0ODg5LzM2NTE0NDQzMy1lYzU2MGE4NS1mNjIyLTQ0ZjgtOGVjZS1hY2ZlYTY1OTNiZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMjA0NDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODUyMTBmOWM0MjMxNzM1M2MzZGMwZGJmMTRlNDZhYWU0YjI2YzgxMjU3ZWZjOWFmNTVjNDQ1OTk2OTdlYzdmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.jZOQUgBULt02JMFndr2Vol-RYNBt7a2r9AxgKk_RGwg"> <img alt="image" width="491" src="https://private-user-images.githubusercontent.com/3684889/365144485-bc9421c4-ae04-4d89-8642-49b960de91ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NTU3ODQsIm5iZiI6MTcyNTY1NTQ4NCwicGF0aCI6Ii8zNjg0ODg5LzM2NTE0NDQ4NS1iYzk0MjFjNC1hZTA0LTRkODktODY0Mi00OWI5NjBkZTkxYWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMjA0NDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTVjODA0YTI2MWE1ZTg3MTA3NzM5YjNlODZkNmRkNjgxYmZjNGJjYWM3YTdjNGMxNTM3MDYwNDk1YjU0OWZkMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.q55gpZsocpzXse6khY7rVzuowbK2NoOmwVBsvPwl1P8">

The exact same sized icon is on the icon page - all indents are respected
There is a similar case in tabbar component, I think this is a problem with icons in general